solver/src/solver/oneinch_solver.rs

```diff
@@ -48,19 +54,17 @@ impl OneInchSolver<GPv2Settlement> {
             "1Inch solver only supported on Mainnet",
         );
 
+        let settlement_address = settlement_contract.address();
         Ok(Self {
-            allowance_fetcher: settlement_contract.clone(),
             settlement_contract,
-            client: Box::new(OneInchClientImpl::default()),
             disabled_protocols: disabled_protocols.into_iter().collect(),
+            client: Box::new(OneInchClientImpl::default()),
+            allowance_fetcher: Box::new(AllowanceManager::new(web3, settlement_address)),
```

Comment:
I guess there is little value in sharing this component across dependencies as the "spender" will always be different?

Reply:
The component can be shared since its tied to the owner (which so far has always been the settlement contract) not the spender. Although caching would have to be spender-specific.